cmds/core/ping/ping_test.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"net"
+	"testing"
+	"time"
+)
+
+type MyConn struct {
+	TestRun string
+}
+
+func (M *MyConn) Read(b []byte) (int, error) {
+	if M.TestRun == "error in read" {
+		return 0, fmt.Errorf("err")
+	}
+	if M.TestRun == "rmsg[0] != ICMP_TYPE_ECHO_REPLY" {
+		b[20] = 0xff
+	}
+	if M.TestRun == "!net6 && cks != cksum(rmsg)" {
+		return 0, nil
+	}
+	b[22] = 0xff
+	b[23] = 0xff
+	return 0, nil
+}
+
+func (M *MyConn) Write(b []byte) (int, error) {
+	if M.TestRun == "error in write" {
+		return 0, fmt.Errorf("err")
+	}
+	return 0, nil
+}
+
+func (M *MyConn) Close() error                       { return nil }
+func (M *MyConn) LocalAddr() net.Addr                { return nil }
```

Comment:
but returning a nil pointer? Maybe, for example, this should return a valid address, e.g. an ip4 with 127.1? I suppose it's fine for now however.

BTW, you may be starting to see the rule that interfaces should be small.